backends/bmv2/jsonconverter.cpp

```diff
@@ -1376,44 +1552,24 @@ JsonConverter::convertTable(const CFG::TableNode* node,
                 int l = slice->getL();
                 mask = Util::maskFromSlice(h, l);
             }
-
-            cstring match_type = mt->name.name;
-            if (mt->name.name == corelib.exactMatch.name) {
+            if (match_type == "valid") {
+                auto mt = refMap->getDeclaration(ke->matchType->path, true)->to<IR::Declaration_ID>();
                 if (expr->is<IR::MethodCallExpression>()) {
                     auto mi = P4::MethodInstance::resolve(expr->to<IR::MethodCallExpression>(),
                                                           refMap, typeMap);
                     if (mi->is<P4::BuiltInMethod>()) {
                         auto bim = mi->to<P4::BuiltInMethod>();
+                        // \TODO: why isn't this handled much earlier? Is it backend specific?
```

Comment:
Yes, this is back-end specific. The representation of the valid fields is completely back-end specific.